### ISSUE TITLE
New Tokens: Code Color Scheme

### ISSUE THREAD

**Issue by bennypowers** [RedHat-UX/red-hat-design-tokens#16]
We need a code syntax color scheme for prism, and alternatively for vim/v****e

**Comment by coreyvickery** [comment]
@bennypowers Does this refer to colors for syntax highlighting?

**Comment by bennypowers** [comment]
yes, so we could use it for markdown highlighting, code playgrounds, snippets, and editor color themes, whathaveyou

**Comment by coreyvickery** [comment]
@bennypowers This is in progress, I am working with other design teams to try and unify one color scheme. It would be great to see examples of the things you need these values for even if they are quick screenshots.

**Comment by bennypowers** [comment]
https://github.com/PrismJS/prism-themes

**Comment by bennypowers** [comment]
I'd recommend starting by defining terminal ascii colours, for example this version of [Catppuccin Mocha for Black Box terminal](https://github.com/catppuccin/blackbox/blob/main/src/Catppuccin-Mocha.json)

from there we can then do a prism syntax theme, treesitter syntax theme (e.g. [that same catppuccin theme for neovim](https://github.com/catppuccin/nvim))

**Comment by coreyvickery** [comment]
@bennypowers I think I can provide you with values because we are moving on to creating `rh-code-block`.

https://github.com/RedHat-UX/red-hat-design-system/issues/328

**Comment by bennypowers** [comment]
I added this to 1.0 milestone, since it's WIP

**Comment by bennypowers** [comment]
We need this for demo playgrounds as well (spin this out into a new issue later on)

**Comment by coreyvickery** [comment]
@bennypowers How would you like me to deliver this information to you? Do you need the hex color codes and token names? Anything else?

**Comment by bennypowers** [comment]
yeah that should do. you're welcome to start a pr if you're interested

**Comment by coreyvickery** [comment]
@bennypowers Color values below and XD is [here](https://xd.adobe.com/view/c4cc943e-618d-4fbe-8ab2-8e29fd135393-919f/).

## Base style (light theme)
| Color name | Purpose | Token | Hex code | Contrast |
| - | - | - | - | - |
| Black | Text | `black-900` | `#151515` | `16.74` |
| Lighter gray | Background | `black-150` | `#F5F5F5` | - |
| Light gray | Border | `black-300` | `#D2D2D2` | - |

## Syntax highlighting (light theme)
| Color name | Purpose | Token | Hex code | Contrast |
| - | - | - | - | - |
| Red | - | `brand-red-600` | `#BE0000` | `6.03` |
| Orange | - | `orange-500` | `#8F4700` | `6.27` |
| Green | - | `light-green-600` | `#486B00` | `5.7` |
| Blue | - | `blue-500` | `#004080` | `9.41` |
| Purple | - | `purple-500` | `#6753AC` | `5.64` |
| Magenta | - | New token |  `#B300B3` | `5.39` |
| Black | Text | `black-900` | `#151515` | `16.74` |
| Gray | Comments and numbers | `black-600` | `#6A6E73` | `4.7` |
| Light gray | Numbers vertical line | `black-300` | `#D2D2D2` | - |
| Lighter gray | Background | `black-150` | `#F5F5F5` | - |

## Text highlighting (light theme)

> Example is [here](https://xd.adobe.com/view/c4cc943e-618d-4fbe-8ab2-8e29fd135393-919f/screen/da512c24-21c7-456d-a511-3486d7a7ef93).

| Color name | Token | Hex code | Contrast |
| - | - | - | - |
| Text color | `black-900` | `#151515` | `12.07`
| Background | `black-300` | `#D2D2D2` | - |

## Tags (light theme)

| Element | Theme |
| - | - |
| Tags | [Light theme](https://xd.adobe.com/view/c4cc943e-618d-4fbe-8ab2-8e29fd135393-919f/screen/869b2d9a-6848-405c-91aa-f7791ece7c26) |

- - -

## Base style (dark theme)
| Color name | Purpose | Token | Hex code | Contrast |
| - | - | - | - | - |
| Light gray | Text | `black-300` | `#D2D2D2` | `10.31` |
| Darker gray | Background | `black-850` | `#212427` | - |

## Syntax highlighting (dark theme)
| Color name | Purpose | Token | Hex code | Contrast |
| - | - | - | - | - |
| Red | - | `brand-red-400` | `#FF442B` | `4.54` |
| Orange | - | `orange-300` | `#EC7A08` | `5.46` |
| Gold | - | `gold-300`| `#F4C145` | `9.32` |
| Green | - | `light-green-400` | `#92D400` | `8.64` |
| Blue | - | `blue-100` | `#BEE1F4` | `11.33` |
| Purple | - | `purple-300` | `#A18FFF` | `5.85` |
| Light gray | Text | `black-300` | `#D2D2D2` | `10.31` |
| Gray | Comments and numbers | `black-500` | `#8A8D90` | `4.67` |
| Dark gray | Numbers vertical line | `black-600` | `#6A6E73` | - |
| Darker gray | Background | `black-850` | `#212427` | - |

## Text highlighting (dark theme)

> Example is [here](https://xd.adobe.com/view/c4cc943e-618d-4fbe-8ab2-8e29fd135393-919f/screen/cac78b18-6e25-4e2e-bf50-b0ebb4fcd9ce).

| Color name | Token | Hex code | Contrast |
| - | - | - | - |
| Text color | `black-300` | `#D2D2D2` | `7`
| Background | `black-800` | `#3C3F42` | - |

## Tags (dark theme)

| Element | Theme |
| - | - |
| Tags | [Dark theme](https://xd.adobe.com/view/c4cc943e-618d-4fbe-8ab2-8e29fd135393-919f/screen/9f71fbc4-b902-49cc-b050-7551a7abdec8) |

**Comment by bennypowers** [comment]
Here are some token (by which i mean programming language token, not design token) names from prism.js, loosely sorted into three groups by order of importance

```
comment
property
keyword
tag
class-name
boolean
constant
punctuation
number
attr-name
string
symbol
variable
operator
attr-value
function
regex

namespace
deleted
selector
char
builtin
inserted
entity
url
atrule
important

prolog
doctype
cdata
```

I get the sense that if we assign token (design tokens) to these, that's pretty much job done.

TODO: demo codepen for Corey to muck about with, featuring:

- HTML/JS/CSS
- Bash
- Markdown
- Go
- Rust
- JSON
- YAML

**Comment by coreyvickery** [comment]
@marionnegp Tagging you to take over.

**Comment by coreyvickery** [comment]
@bennypowers @nikkimk Current colors, see screens `3` and `9`.

https://xd.adobe.com/view/c4cc943e-618d-4fbe-8ab2-8e29fd135393-919f/

Let me know if you have specific questions about where to apply them.

**Comment by bennypowers** [comment]
are those existing crayon colours?

**Comment by markcaron** [comment]
Feel free to disagree, but as for the language support, I'd say...

### Need first

1. HTML/JS/CSS
2. Bash
3. Typescript
4. Markdown
5. JSON
6. YAML

### Next set

1. PHP
2. Python
3. Java
4. Twig
5. C#
6. Go
7. Rust

_And anything else forgetting._

**Comment by coreyvickery** [comment]
@markcaron Closing this because our Felt tokens work will aim to solve this. I mentioned this in our working group Slack channel to keep a record.
